### ISSUE TITLE
[suggestion] Reset with new group

### ISSUE THREAD

**Issue by tflo** [Wetxius/Numeration#6]
I would prefer that data would also be reset (or asked to be reset, if `silent_reset` is not set) when I join a _group_, not only for new _instances_.

What do you think?

**Comment by Wetxius** [comment]
How to check if it new group?

**Comment by tflo** [comment]
No idea. (I’m not familiar with the API at all.) But either Skada or Recount or Details (or all of them) do that.
I’ll check it out, give me a minute or two.

**Comment by tflo** [comment]
- Recount: Both options (group or/and instance)
- Details: I gave up searching for the option; the UI is such an incredibly and ugly pile of mess
- Skada: Also offers both: reset on instance or/and group

Do you want me to find out how they do that? But, I guess, you are more proficient with that.

**Comment by Wetxius** [comment]
Check this https://gist.github.com/Wetxius/67e0f5a14fc269d25ac9ec800ac3c5b6
Code from Skada.

**Comment by tflo** [comment]
Going to test it with some Honey Smasher/Harvester groups.

**Comment by tflo** [comment]
Nope. It didn’t reset.

More precisely: I accepted an invite, i.e. became member of a group, and Overall data was not reset in display.

**Comment by tflo** [comment]
OK, doing another test, this time with `silent_reset` off, to be sure.

**Comment by Wetxius** [comment]
It won't work if you reload/enter game and was not in group before. It should reset only for current game when you was and group, leave and find new. About overall it's bug causing old function code lost as I check by same event.

**Comment by tflo** [comment]
Now it has reset as I created a group myself.

**Comment by tflo** [comment]
> About overall it's bug causing old function code lost as I check by same event.

Sorry, don’t understand this.

**Comment by tflo** [comment]
I’ll test your new Core.lua with other toons tonight and tomorrow. So, [the things you described](https://github.com/Wetxius/Numeration/issues/6#issuecomment-721903111) and I didn’t fully understand should be ruled out. I’ll report back.

**Comment by tflo** [comment]
Worked fine now. Let me do another one with another toon .

**Comment by tflo** [comment]
No, this time it didn’t work. I was definitely not in a group (freshly logged-in), then joined a group --> no Reset dialog and no reset.

PS:

On another toon, also freshly logged-in, then accepted invite --> Reset dialog.

So, ATM, I don’t know what it is that makes it work and what not. 😕

**Comment by tflo** [comment]
Next try: Several group invites (non-instance): All worked, i.e. data was reset / asked to be reset. This was all on one toon.

So, I don’t know. Seems quite stable. One exception, see above (was on a different t toon); maybe just ignore it? User fault, whatever?

I would release it. Even if it does not work 100/100, it will still be better than no group reset. Just my opinion.

**Comment by tflo** [comment]
Still: It mostly works, but sometimes not. Unable to figure out the conditions. ATM.

**Comment by Wetxius** [comment]
Check new version https://gist.github.com/Wetxius/67e0f5a14fc269d25ac9ec800ac3c5b6

**Comment by tflo** [comment]
Thanks, I’ll report back, but maybe not today.

**Comment by tflo** [comment]
Sorry for not having reported back that long. With the Icecrown rares I was mainly using Recount, since it shows me also ungrouped players. (Sidenote: I do _not_ want to see this feature in Numeration. Numeration should stay as lightweight as it is, IMO.)

But from the most recent Honey Harvester open world groups:

As you already mentioned (if I got it right), after logging in it will – in most cases – _not_ reset on the first group I join. However, I’m pretty sure I’ve also seen it reset on the first group after login; maybe it was because I logged in, then went to another zone, then joined a group. Is this possible?

And: I’ve also seen it _not_ reset with the second group after login. I already thought, it now stopped resetting completely, but it then always reset with the third group in these cases. (This was 2 or 3 times; zone changes _may_ have been involved.)

**Comment by Wetxius** [comment]
Nope, it's independent of zone changes.
Check another way https://gist.github.com/Wetxius/67e0f5a14fc269d25ac9ec800ac3c5b6
Should always reset even after first login.

**Comment by tflo** [comment]
Thanks, have copied the new Core.lua, and will report back; hopefully sooner than the last time.

A suggestion for future patches via pastebin: maybe add a little identifier to the code, for example a time stamp at the beginning:

```
-- version 20201121T2117
```
or something like that. Would help me to make sure I’m on the correctly (most recently) patched file. (That is because I’m still patching around in other files, and sometimes I reload the whole add-on from the curseforge repo to my local one.)

**Comment by tflo** [comment]
OK, one test so far: Icecrown rares: logged-in into toon, joined raid group --> add-on asked me to reset.

Seems fine. Will do some more tests.

**Comment by tflo** [comment]
Edit to the above: got several reset dialogs while still being in the same raid group and waiting for the rare. So, something goes wrong, it seems.

**Comment by tflo** [comment]
OK, same thing with the next toon: logged-in, joined raid group, got reset dialog, then got immediatly 2 or 3 more reset confirmation dialogs. Then after ~40s waiting two more dialogs in a row. ~20s later one more. ~60s later one more.

Not sure what could be triggering this. Maybe party changes within the raid group?

PS: 
Some more reset dialogs after the fight, while still being in the group.
So, I will discard your latest patch for the moment.
